Export Improvements

mtrive opened the pull request:
**Problem**
Project Auditor evolved from simply reporting code and settings issues to also list shaders and shader variants, which have their own custom properties such as number of variants/passes/keywords, etc.... These are shown correctly in the UI because each view has its own layout. Unfortunately, the _Export_ functionality does not have any visibility on UI layout so it does not export any issue specific properties.

**Solution**
Define a data layout for each issue category, which can be used by both the UI and the _Export_ feature.
With this change, the way you Export also changes. Instead of having a single Export button that generates a csv file containing all issues, each view has its own Export button.

![view-export](https://user-images.githubusercontent.com/12098182/109493044-d226e180-7a83-11eb-89fe-a7308bc7fec8.PNG)

This means, each view will need to be exported individually.
However, with this change, Shader Variants window will also have its own Export button.

![variants-export](https://user-images.githubusercontent.com/12098182/109493072-db17b300-7a83-11eb-94dd-691be4fb6107.PNG)

unitystevem reviewed (changes requested):
AnalysisView.cs - default column sizes seem to have changed to a much more simplified, uniform view. As a result, the Path column in Assets is way too small by default, and in the Code window, Issue and Area are too big, Filename is a bit too small.

Would be nice to still have a single button that can export all the separate views to different CSV files all at once. If that would be too confusing to add to the UI, could the file dialog box to choose where to save the CSV in each view at least remember the last folder you selected? It keeps defaulting back to the root of the project, so navigating from there to the chosen folder 4 times in a row when exporting all the views is pretty annoying.

Code CSV exports with “Severity” listed as either “Default” or “Warning”. I think this would be better with a column header of “Critical Code Path” and values of “true” or “false”.

Some exports appear to include issues that aren’t visible in the Project Auditor window itself. For instance, in my test project, Player Settings exports "Player: Prebake Collision Meshes","BuildSize|LoadTimes”, an issue that isn’t shown in the Editor window. Could this be a bug in the Area filtering code in the views? (Note how this issue covers two areas with "BuildSize|LoadTimes”)

In the Shader CSV export, the first column has no title and values consisting only of “Default”. This seems redundant and should be removed.

Unrelated to this branch (so consider this optional, at least in this branch), but the initial window text (before clicking analyse) is out of date, says there’s only two tabs, instead of several items on a drop-down menu.

mtrive commented:
1. Path column size: fixed
2. Export All: Yes, I am a bit concerned it will be confusing but I would like to understand this better. Would you need this for project reviews or you think the end-user would need this? The reason I am asking is that, once this PR lands, I will start implementing a global Load/Save functionality, which will save everything. Note, this will likely be using _json_ since _csv_ is not a suitable format for this purpose. Would you be happy with this (instead of the Export All views)?
3. Export Directory: I fixed this so it remembers the last export directory
4. Critical code path: The intention was to use the Severity property but on a second thought I agree that probably does not map well. Fixed.
5. Issues affecting multiple times: oh dear, this is not the first time this breaks. I just fixed it (also in `master`). Thanks for pointing this out.
5. Shaders severity column: this was intended to show whether the shader failed failed to compile. I agree that, if all shaders compile fine, the column does not make sense as is. I will disable it for the time being, then we can re-evaluate how/if we want to bring it back.
6. Updated intro text: I agree. Let's update that separately.

Thanks Steve!

unitystevem commented:
All looks good. To answer your export question: I suppose exporting all views at once is a workflow that would probably only come up during Project Reviews and/or testing the tool rather than during normal customer use. Now that Project Auditor remembers the last export directory, the workflow feels a lot better, even if it's still a lot of clicking to export everything.

When working remotely with a customer project it's basically fine now. I think an "export all views" button might still be useful for when we get back to onsite Project Reviews, just to make it quick to save all the data to go over for the report. Load/Save would address this issue too, of course.
